var/spack/repos/builtin/packages/daos/package.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Daos(SConsPackage):
+    """The Distributed Asynchronous Object Storage (DAOS) is an open-source
+    software-defined object store designed from the ground up for massively
+    distributed Non Volatile Memory (NVM)."""
+
+    homepage = "https://github.com/daos-stack/daos"
+    git = "https://github.com/daos-stack/daos.git"
+    maintainers("hyoklee")
+
+    version("master", branch="master", submodules=True)
+    version("2.2.0", tag="v2.2.0", submodules=True)
+    variant(
+        "debug", default=False, description="Enable debugging info and strict compile warnings"
+    )
+
+    depends_on("argobots")
+    depends_on("boost@develop+python", type="build")
+    depends_on("cmocka", type="build")
+    depends_on("dpdk@22.11:")
+    depends_on("go", type="build")
+    depends_on("hwloc")
+    depends_on("isa-l")
+    depends_on("isa-l_crypto")
+    depends_on("libfabric")
+    depends_on("libfuse@3.6.1:")
+    depends_on("libuuid")
+    depends_on("libunwind")
+    depends_on("libyaml")
+    depends_on("mercury+boostsys")
+    depends_on("mpich")
+    depends_on("openssl")
+    depends_on("pmdk")
+    depends_on("protobuf-c")
+    depends_on("py-distro")
+    depends_on("readline")
+    depends_on("scons@4.4.0:")
+    depends_on("spdk@23.01:+shared+rdma")
```

Comment:
also just a warning that 23.01 is not officially supported by daos yet, see https://github.com/daos-stack/daos/blob/master/utils/build.config that tells you which version is supported